Grbl_Esp32/src/PinSettingsDefinitions.cpp

```diff
@@ -0,0 +1,960 @@
+// This file attempts to build all the pin settings definitions, while retaining backward compatibility with the
+// original machine files.
+//
+// Note that the source code shouldn't use #define'd pins directly, but always use the settings in the code.
+// To aid that, I do NOT want to include the default <Arduino.h> from this file (indirectly). Instead, this
+// file defines the GPIO_NUM_* and I2SO(*) as macro's to strings that the Pin class understands, and uses that.
+//
+// To make sure we don't accidentally ripple macro's some place they don't belong, I tried to do this as isolated
+// as possible.
+//
+// NOTE: The order in which defines and includes are used here matters!
+//
+// All the preprocessor things are handled first, followed by the real code.
+
+// First do the defines that take care of GPIO pin name mapping:
+
+#include <cstdint>
+#include "Config.h"
+
+// Include the file that loads the machine-specific config file.
+// machine.h must be edited to choose the desired file.
+#include "Machine.h"
+
+// machine_common.h contains settings that do not change
+#include "MachineCommon.h"
+
+// Update missing definitions with defaults:
+#include "Defaults.h"
+
```

Comment:
Need pinsettings for:
#define I2S_OUT_BCK     GPIO_NUM_22
#define I2S_OUT_WS      GPIO_NUM_17
#define I2S_OUT_DATA    GPIO_NUM_21

Also need the appropriate changeds in I2Sout

Reply:
@MitchBradley The reason I was in doubt was because I was unsure of the initialization order. I _think_ it should work, but i prefer to be sure. 
I'll change I2S to I2SO. We could also wire it dynamically, based on the attributes specified in code (Attributes::Input / Attributes::Output). I'm fine either way.